### ISSUE TITLE
Caret browsing issues with Qt 5.10

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#3523]
Looks like various tests fail on AppVeyor: https://ci.appveyor.com/project/qutebrowser/qutebrowser/build/master-3165/job/ge2hps2livqrb0mx#L49

Can't test on Travis because of #3522, but locally they seem to pass on Linux (but some are flaky?).

@artur-shaik If you find the time, can you take a look at this?

**Comment by artur-shaik** [comment]
Like a challenge... Qt5.10 on Windows 😨

**Comment by The-Compiler** [comment]
You should be able to `pip install PyQt5==5.10` on Windows FWIW.

**Comment by The-Compiler** [comment]
Hmm, that's odd. I also get issues on Archlinux, but only via the Docker container, not my native install:

```
$ docker run -it --rm --privileged -v "$PWD:/outside" -e QUTE_BDD_WEBENGINE=1 -e DOCKER=archlinux-webengine -e CI=1 -e TRAVIS=1 qutebrowser/travis:archlinux-webengine
```

Though I don't think it's always the same tests failing, maybe something just got flaky...

**Comment by The-Compiler** [comment]
This seems to happen because the caret code is now using `selectedText()` instead of JS to get the selection on Qt 5.10. Apparently, that's still not working quite like it should, but I currently don't want to debug that :wink:

I fixed this by always using JS - for the foreseeable future we'll need to have a callback-based interface for `selection()` anyways, so it doesn't hurt. I also got rid of `has_selection()` entirely as all it does in QtWebEngine is checking whether the text is empty.
